presto-hive/src/main/java/com/facebook/presto/hive/HiveUtil.java

```diff
@@ -239,6 +242,23 @@ public static void setReadColumns(Configuration configuration, List<Integer> rea
         configuration.setBoolean(READ_ALL_COLUMNS, false);
     }
 
+    public static Optional<CompressionCodec> getCompressionCodec(TextInputFormat inputFormat, Path file)
+            throws NoSuchFieldException, IllegalAccessException
+    {
+        if (inputFormat == null || file == null) {
+            return Optional.empty();
+        }
+
+        Field field = inputFormat.getClass().getDeclaredField("compressionCodecs");
```

Comment:
Change this to be a constant (`private static final`) initialized in a static initializer block and fail with `AssertionError` if the field does not exist. See `ParquetRecordWriterUtil` for an example.

Reply:
Refactored